var/spack/repos/builtin/packages/libfabric/package.py

```diff
@@ -119,6 +123,7 @@ class Libfabric(AutotoolsPackage):
     depends_on("ucx", when="@1.18.0: fabrics=ucx")
     depends_on("uuid", when="fabrics=opx")
     depends_on("numactl", when="fabrics=opx")
+    depends_on("liburing@2.1:", when="+uring")
```

Comment:
Libfabric already enables uring support by default if the right libraries are available (with an option to explicitly disable it).  If it is not available, it is silently disabled. You are likely running into just missing deps. Do you want a new variant for it, or just add the dependency?

Ref: https://github.com/ofiwg/libfabric/blob/b3963556a41c2aeb2b4682a0d149550c09a65efa/configure.ac#L588-#L598

Reply:
It could go either way. I made it an explicit variant so the dependency would be optional and turned it off by default because there is a kernel version requirement for io_uring. The feature is also turned off by default in the code unless `FI_TCP_IO_URING=1` is set.